cob_cartesian_controller/src/trajectory_profile_generator/trajectory_profile_generator_ramp.cpp

```diff
@@ -0,0 +1,155 @@
+/*!
+ *****************************************************************
+ * \file
+ *
+ * \note
+ *   Copyright (c) 2015 \n
+ *   Fraunhofer Institute for Manufacturing Engineering
+ *   and Automation (IPA) \n\n
+ *
+ *****************************************************************
+ *
+ * \note
+ *   Project name: care-o-bot
+ * \note
+ *   ROS stack name: cob_control
+ * \note
+ *   ROS package name: cob_cartesian_controller
+ *
+ * \author
+ *   Author: Christoph Mark, email: christoph.mark@ipa.fraunhofer.de / christoph.mark@gmail.com
+ *
+ * \date Date of creation: September, 2015
+ *
+ * \brief
+ *
+ ****************************************************************/
+
+#include "ros/ros.h"
+#include "cob_cartesian_controller/trajectory_profile_generator/trajectory_profile_generator_ramp.h"
+/* BEGIN TrajectoryProfileRamp ********************************************************************************************/
+inline cob_cartesian_controller::ProfileTimings TrajectoryProfileRamp::getProfileTimings(double Se, double te, double accl, double vel, bool calcMaxTe)
+{
+    CartesianControllerUtils utils;
+    cob_cartesian_controller::ProfileTimings pt;
+    int steps_te, steps_tv, steps_tb = 0;
+    double tv, tb = 0.0;
+
+    // Calculate the Sinoid Profile Parameters
+    if (vel > sqrt(std::fabs(Se) * accl))
+    {
+        vel = sqrt(std::fabs(Se) * accl);
+    }
+
+    if(vel > 0.00001)
+    {
+        tb = utils.roundUpToMultiplier(vel / accl, params_.profile.t_ipo);
+        if(calcMaxTe)
+        {
+            te = utils.roundUpToMultiplier((std::fabs(Se) / vel) + tb,  params_.profile.t_ipo);
+        }
+        tv = te - tb;
+
+        // Interpolationsteps for every timesequence
+        pt.tb = tb;
+        pt.tv = tv;
+        pt.te = te;
+
+        pt.steps_tb = pt.tb/params_.profile.t_ipo;
+        pt.steps_tv = (pt.tv-pt.tb)/params_.profile.t_ipo;
+        pt.steps_te = (pt.te-pt.tv)/params_.profile.t_ipo;
+
+        pt.vel = vel;
+        pt.ok = true;
+    }
+    else
+    {
+        pt.ok = false;
+    }
+
+    return pt;
+}
+
+
+inline bool TrajectoryProfileRamp::generatePath(cob_cartesian_controller::PathArray &pa)
+{
+    std::vector<double> array;
+    cob_cartesian_controller::ProfileTimings pt;
+    double accl_max = params_.profile.accl;
+    double vel_max = params_.profile.vel;
+
+    // Calculate the Profile Timings
+    pt = getProfileTimings(pa.Se_, pt_max_.te, accl_max, vel_max, false);
+    if(pt.ok)
+    {
+        array = getTrajectory(pa.Se_, accl_max, pt.vel, params_.profile.t_ipo, pt.steps_tb, pt.steps_tv, pt.steps_te, pt.tb, pt.tv, pt.te);
+    }
+    else
+    {
+        array.push_back(0);
+    }
+
+    pa.array_ = array;
+    return true;
+}
+
+inline std::vector<double> TrajectoryProfileRamp::getTrajectory(double se, double accl, double vel, double t_ipo, double steps_tb, double steps_tv, double steps_te, double tb, double tv, double te)
+{
+    std::vector<double> array;
+    unsigned int i = 1;
```

Comment:
This function interpolates the path for time > 0 (here in discrete time i > 0) ... i = 0 would represent the actual position of the tracking_frame which can be neglected.